rebase.go

```diff
@@ -69,6 +71,59 @@ func newRebaseOperationFromC(c *C.git_rebase_operation) *RebaseOperation {
 	return operation
 }
 
+//export commitSignCallback
+func commitSignCallback(_signature *C.git_buf, _signature_field *C.git_buf, _commit_content *C.char, _payload unsafe.Pointer) C.int {
+	opts, ok := pointerHandles.Get(_payload).(RebaseOptions)
+	if !ok {
+		panic("invalid sign payload")
+	}
+
+	if opts.SigningCallback == nil {
+		return C.GIT_PASSTHROUGH
+	}
+
+	commitContent := C.GoString(_commit_content)
+
+	signature, signatureField, err := opts.SigningCallback(commitContent)
+	if err != nil {
+		return C.int(-1)
+	}
+
+	fillBuf := func(bufData string, buf *C.git_buf) error {
+		clen := C.size_t(len(bufData))
+		cstr := unsafe.Pointer(C.CString(bufData))
+
+		// over-assign by a byte (see below)
+		if int(C.git_buf_grow(buf, clen+1)) != 0 {
+			return errors.New("could not grow buffer")
+		}
```

Comment:
`git_buf_grow()` internally calls `git__realloc`, which calls [`realloc(3)`](https://linux.die.net/man/3/realloc), which says that "If the new size is larger than the old size, the added memory will _not_ be initialized.". This means that it's not guaranteed that the very last byte in the buffer is going to be a null byte after L107.

Would it make sense to call [`git_buf_sanitize`](https://github.com/libgit2/libgit2/blob/c71321a099373753c22055921c8f538afed5ebb6/src/buffer.c#L143-L150) to ensure the null byte is there?

Reply:
I did have this doing memset() to null bytes but figured that it was probably done by libgit internally, apparently not...I think I'll just make it do that again to make sure that it has a null byte and to avoid any issues with sizes being incorrect